app/static/js/bonnerManagement.js

```diff
@@ -14,14 +14,21 @@ function cohortRequest(year, method, username){
   })
 }
 
+function addSearchCapabilities(inputElement){
+    inputElement = $(inputElement);
+    inputElement.on("input", function(){
+        let year = $(this).data('year');
+        searchUser(this.id, s => cohortRequest(year, "add", s.username), false, null, "student");
+    });
+}
+
+
 /*** Run After Page Load *************************************/
 $(document).ready(function(){
-    $("input[type=search]").on("input", function(){
-        let year = $(this).data('year')
-        searchUser(this.id, s => cohortRequest(year, "add", s.username), false, null, "student")
-    });
+    $("#addCohort").on('click', addCohort);
+    $("input[type=search]").each((i, e) => addSearchCapabilities(e));
     $(".removeBonner").on("click",function(){
-        let year = $(this).data('year')
+        let year = $(this).data('year');
```

Comment:
I like that this function now completely has semicolons for line endings but it seems like it is a little inconsistent in other locations where you have written JS code. I would recommend trying to either keep it consistent or just not worry too much about it in areas where you may not have made a change.